relayer/src/link/pending.rs

```diff
@@ -177,16 +178,28 @@ impl<Chain: ChainHandle> PendingTxs<Chain> {
                         // relayer to resubmit the transaction to the chain again.
                         error!("timed out while confirming {}", tx_hashes);
 
-                        let resubmit_res = resubmit(pending.original_od.clone());
-
-                        match resubmit_res {
-                            Ok(reply) => {
-                                self.insert_new_pending_tx(reply, pending.original_od);
-                                Ok(None)
+                        match resubmit {
+                            Some(f) => {
+                                let new_od = relay_path
+                                    .regenerate_operational_data(pending.original_od.clone());
+                                match new_od.map(f) {
```

Comment:
The code here is a bit cryptic, can we add just a couple of remarks on what is happening and why?